File: src/cpp/src/hello.cpp
#include "hello.h"
#include <iostream>

namespace hello {

Greeter::Greeter(const std::string& name) : name_(name) {}

std::string Greeter::greet() const {
    return "Hello, " + name_ + "!";
}

std::string Greeter::farewell() const {
    return "Goodbye, " + name_ + "!";
}

}  // namespace hello

File: src/python/hello.py
"""Hello World Python Module."""

__version__ = "0.1.0"


def greet(name: str = "World") -> str:
    """Return a greeting message."""
    return f"Hello, {name}!"


def farewell(name: str = "World") -> str:
    """Return a farewell message."""
    return f"Goodbye, {name}!"


if __name__ == "__main__":
    print(greet())
    print(farewell())
